code/data_structures/src/list/circular_linked_list/circular_linkedlist.c

```diff
@@ -0,0 +1,713 @@
+#include <stdio.h>
+
+#include <stdlib.h>
+
+ 
+
+struct node
+
+{
+
+    int data;
+
+    struct node *link;
+
+};
+
+ 
+
+struct node *head = NULL, *x, *y, *z;
+
+ 
+
+void create();
+
+void ins_at_beg();
+
+void ins_at_pos();
+
+void del_at_beg();
+
+void del_at_pos();
+
+void traverse();
+
+void search();
+
+void sort();
+
+void update();
+
+void rev_traverse(struct node *p);
+
+ 
+
+void main()
```

Comment:
_int_. also, the type for all functions goes on its own line. Follow the [style guide](https://github.com/OpenGenus/cosmos/blob/master/guides/coding_style/c/README.md).